demux/demux_libarchive.c

```diff
@@ -76,7 +76,9 @@ static int open_file(struct demuxer *demuxer, enum demux_check check)
     struct playlist *pl = talloc_zero(demuxer, struct playlist);
     demuxer->playlist = pl;
 
-    char *prefix = mp_normalize_path(NULL, mp_url_escape(mpa, demuxer->stream->url, "~|%"));
+    char *normalized = mp_normalize_path(NULL, demuxer->stream->url);
+    char *prefix = mp_url_escape(NULL, normalized, "~|%");
+    talloc_free(normalized);
```

Comment:
could just swap the ptr ownership around, no?
```suggestion
    char *prefix = mp_url_escape(NULL, mp_normalize_path(mpa, demuxer->stream->url), "~|%");
```